### PR TITLE
[Quest] Update quest format for AF SCH

### PR DESCRIPTION
**Pull request by Chronos77**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Update quest file format for AF SCH

### CODE REVIEW

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (claywar) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua]: This could be simplified to:
`if npcUtil.tradeHasExactly(trade, {{ xi.items.VELLUM, 12 }}) then`

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (claywar) [inline scripts/quests/crystalWar/On_Sabbatical.lua]: Probably don't need the condition here, but it doesn't hurt

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (claywar) [inline scripts/quests/crystalWar/On_Sabbatical.lua]: There's no need to lookup the player's zone here, unless you're planning on making a more generic function later.  This can only be triggered in Pashhow, therefore we can just use `local ID = zones[xi.zone.PASHHOW_MARSHLANDS_S]`

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (claywar) [inline scripts/quests/crystalWar/On_Sabbatical.lua]: This is not necessary, as quest:complete() will fail if the item cannot be obtained.  Can just use the below else condition

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (Chronos77) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua]: done

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (Chronos77) [reply scripts/quests/crystalWar/On_Sabbatical.lua]: done

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (Chronos77) [reply scripts/quests/crystalWar/On_Sabbatical.lua]: done

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (Chronos77) [reply scripts/quests/crystalWar/On_Sabbatical.lua]: done

**Comment by TeoTwawki** [comment]
> [Chronos77](https://github.com/Chronos77) requested your review on this pull request.

Out of my wheelhouse, I defer to Clay and Xaver. (I am presently catching up on my github notifications, better late than never!)

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (Xaver-DaRed) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua]: We generaly use `Prog` for the variable name on quests to track progress, for consistency.

Comment (claywar) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua]: Correct, `Prog` for quests, `Status` for missions (since missions depend on missionStatus bindings)

File: scripts/zones/The_Eldieme_Necropolis_[S]/npcs/Erlene.lua
Comment (Xaver-DaRed) [inline scripts/zones/The_Eldieme_Necropolis_[S]/npcs/Erlene.lua]: This could go to defaultAction
They play better with this system aswell

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (claywar) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua]: Even though scripts/globals/settings is empty, we've been using the pattern of still adding the require

**Review by Xaver-DaRed — COMMENTED** [review]
LGTM, just those 2 minor things:
- The default action
- The variable name, from `Status` to `Prog`

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (claywar) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua]: npcUtil uses confirmItem, better practice to use confirmTrade (and would prefer at the event finish of event 12)

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (claywar) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua]: We shouldn't be reading this variable over and over, I would suggest local questProgress = quest:getVar(player, 'Status') and then compare that

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (claywar) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua]: Unnecessary, since this only occurs here.  Define local eldiemeID = require(...) at the top of the file

File: scripts/quests/crystalWar/Downward_Helix.lua
Comment (claywar) [inline scripts/quests/crystalWar/Downward_Helix.lua]: Check out the usage of VanadielUniqueDay() in other quests for day wait timers

File: scripts/quests/crystalWar/Downward_Helix.lua
Comment (claywar) [inline scripts/quests/crystalWar/Downward_Helix.lua]: Similar to a previous comment, let's avoid repeated reads and use a variable

File: scripts/quests/crystalWar/Downward_Helix.lua
Comment (claywar) [inline scripts/quests/crystalWar/Downward_Helix.lua:100]: 👍

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (claywar) [inline scripts/quests/crystalWar/On_Sabbatical.lua]: Similar to previous comment

File: scripts/quests/crystalWar/On_Sabbatical.lua
Comment (claywar) [inline scripts/quests/crystalWar/On_Sabbatical.lua]: Prefer == 2 and use event 20, instead of a ~=

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (Xaver-DaRed) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua:10]: require('scripts/globals/settings') is unneeded

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (Chronos77) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua]: I would like to but...
image

Comment (claywar) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua]: Yep, that's on me after that pattern was changed recently and didn't update this issue.  You're good here 👍

**Comment by Chronos77** [comment]
Conflict resolved

**Review by zach2good — APPROVED** [review]
Pending @claywar sign-off

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (claywar) [inline scripts/quests/crystalWar/A_Little_Knowledge.lua:185]: We still need this replaceDefault event if the below block is not applied

**Comment by zach2good** [comment]
A few things have changed since this was last rebased, here's some new CI dings:
```
Run touch lua_checks.txt
Checking scripts/quests/crystalWar/A_Little_Knowledge.lua 1 warning

    scripts/quests/crystalWar/A_Little_Knowledge.lua:201:47: (W113) accessing undefined variable eldiemeID

Use of deprecated/unnecessary require: scripts/globals/items. This should be removed: scripts/quests/crystalWar/Downward_Helix.lua:9
require('scripts/globals/items')                              <-- HERE

Lua styling errors: 1
Use of deprecated/unnecessary require: scripts/globals/items. This should be removed: scripts/quests/crystalWar/On_Sabbatical.lua:10
require('scripts/globals/items')                              <-- HERE

Lua styling errors: 1
Error: Process completed with exit code 1.
```

**Review by claywar — COMMENTED** [review]
Seems reasonable, just one item to address

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (Chronos77) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua:185]: Done

Comment (claywar) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua:185]: I seem to remember this needing more than just completed quest.  Isn't there something along the lines of either having to be SCH, or having SCH at least to level 5?

**Review by claywar — APPROVED** [review]
Seems reasonable, just one thing about the requirements for getting those two spells after completing A Little Knowledge

File: scripts/quests/crystalWar/A_Little_Knowledge.lua
Comment (Chronos77) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua:185]: the canLearnSpell does not do the check?

Comment (claywar) [reply scripts/quests/crystalWar/A_Little_Knowledge.lua:185]: Yep, you're absolutely correct

**Pull request merged**
